src/store/providers/actionProvider.tsx

```diff
@@ -0,0 +1,20 @@
+import { useState } from "react";
+import { ActionState } from "../../types";
+import { ActionContext } from "../context";
+
+export const ActionProvider = ({ children }: { children: unknown }) => {
```

Comment:
Use interfaces as props and explicitly declare ActionProvider as React.FC. This is also useful for code clarity